BMS Workspace-git/bms-1227-fw/Phantom_drivers/drivers/include/can_charger_driver.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * can_driver.h
+ *
+ *  Created on: Jun 14, 2024
+ *      Author: Mohnish Devarapalli
+ */
+
+#ifndef PHANTOM_DRIVERS_CAN_DRIVER_H_
+#define PHANTOM_DRIVERS_CAN_DRIVER_H_
+
+
+#define CAN_Protocol_num 1000;
+#define CAN_baud_rate 250000;
+#define CAN_cycle 1000;// in ms ei. 1 second cycle
+
+
+int min_voltage = 2976; // 297.6 V
+int max_voltage = 4032; // 403.2 V
+int single_battery_max_protection_voltage = 4200; // 4.2 V
+int single_battery_min_voltage = 3100; // 3.1 V
+int single_battery_min_protection_voltage  = 3000; // 3V
+int min_current = 0; // 0A
+int max_current = 26; //2.6A
+int max_discharge_current = 35;// 35A
+int AH = 26; //amp hours 2.6 AH
+int actual_charging_current = 0; // TODO find actual charging current, not in data sheet
+int max_temp = 60-100;// 60 C
+int min_temp = 0-100; // 0 C
+int battery_numbers = 96; // 96 batteries
+int id1 = 0x1806E6F4;
+
+
+typedef struct can_message1{
+    uint8_t byte1;
+    uint8_t byte2;
+    uint8_t byte3;
+    uint8_t byte4;
+    uint8_t byte5;
+    uint8_t byte6;
+    uint8_t byte7;
+    uint8_t byte8;
+    uint32_t id;
+}can_message1;
+
+
+
+void charger_init(){
+
+    can_message1 message_10;
+    can_message1 message_11;
+    can_message1 message_12;
+    can_message1 message_13;
+    can_message1 message_14;
+
+    message_10.id = id1;
```

Comment:
Make messages 10 to 14 have global variable pointers, then initialize them in this function. We need to be able to access these CAN messages outside of the init function, and right now they'd be deleted at the end of it.